sources/updater/updog/src/bin/updata.rs

```diff
@@ -142,25 +142,45 @@ struct WaveArgs {
     arch: String,
 
     // start bound id for this wave (0 <= x < 2048)
-    #[structopt(short = "b", long = "bound-id")]
-    bound: u32,
+    #[structopt(
+        short = "b",
+        long = "bound-id",
+        requires = "start",
+        conflicts_with = "wave_file"
+    )]
+    bound: Option<u32>,
 
     // start time for this wave
-    #[structopt(short = "s", long = "start-time")]
+    #[structopt(
+        short = "s",
+        long = "start-time",
+        requires = "bound",
+        conflicts_with = "wave_file"
+    )]
     start: Option<DateTime<Utc>>,
+
+    // file that contains wave structure
+    #[structopt(short = "w", long = "wave-file", conflicts_with_all = &["bound", "start"])]
+    wave_file: Option<PathBuf>,
 }
 
 impl WaveArgs {
     fn add(self) -> Result<()> {
         let mut manifest: Manifest = update_metadata::load_file(&self.file)?;
-        let start = self.start.context(error::WaveStartArg)?;
-        let num_matching = manifest.add_wave(
-            self.variant,
-            self.arch,
-            self.image_version,
-            self.bound,
-            start,
-        )?;
+
+        let num_matching = if self.wave_file.is_none() {
+            let bound = self.bound.context(error::WaveBoundArg)?;
+            let start = self.start.context(error::WaveStartArg)?;
```

Comment:
Seems kind of weird to me that when adding waves using TOML files, waves are defined by `end_time`'s, but when adding a single wave, it's specified using a single `start_time`. Are they functionally the same, or is adding a single wave applicable for the first wave only?
I have the same question for `add_wave` and `add_waves` in `update_metadata`.

Reply:
I think this comes from different views of what makes a wave. It's probably most accurate to say that update waves are an implicit result of the "bounds" added via, for example, add_wave.
For example, say you add a bound at 1024 with a start time in 24 hours. Then you have two waves;
- clients from 0 -1023 starting now and ending in 24 hours
- clients from 1024-2048 starting in 24 hours

Reply:
I originally had the keys in the TOMl `end_seed` and `start_time`, but that confused me a bit when reasoning through it.

Reply:
Could `start_time` then be changed to `end_time`, and `bound` be changed to `end_seed`  in that case? For naming consistency?

If we're keeping the the terms as they are, then I think a clarify comment somewhere here or above `add_wave` will help.

Reply:
I think some of the inconsistency comes from the different contexts of writing and reading a config file vs. reading code. I think the format of the config file makes sense but I agree that the inconsistency in code is bad...

Reply:
As per #881 - this format has been changed.